.github/workflows/node-build.yml

```diff
@@ -84,6 +99,9 @@ jobs:
     timeout-minutes: 5
     steps:
       - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
```

Comment:
we should only recursively fetch for the action jobs/steps that require the specs to be loaded, which should only be `auth` `backend` `integration-test` and `performance-test`